### PR TITLE
Aaron/benchmark

### PR DESCRIPTION
**Pull request by atucker**
This PR adds an implementation of hyperoperations to the benchmarks, expressed using Knuth's up arrow notation. Hyperoperations generalize the way that multiplication is repeated addition. This adds exponentiation, but also repeated exponentiation, and repeated repeated exponentiation, etc.

https://en.wikipedia.org/wiki/Knuth%27s_up-arrow_notation

### CODE REVIEW

**Comment by sampsyo** [comment]
Very nifty idea! Looks great, and potentially very expensive. :smiley:

File: benchmarks/up-arrow.bril [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+@up_arrow(num: int, arrows: int, repeats: int): int {
+  one: int = const 1;
+
+  ans: int = id num;  
+  i: int = const 1;
+
+.loopstart:
+  keepgoing: int = lt i repeats;
```
Comment (Pat-Lafon) [inline benchmarks/up-arrow.bril:15]: `int` -> `bool`?

Comment (atucker) [reply benchmarks/up-arrow.bril:15]: Oh, good catch! Yeah brili still lets this work for some reason...

Comment (Pat-Lafon) [reply benchmarks/up-arrow.bril:15]: Yeah, different implementations. I'm just trying to make sure `brilirs` runs on all of the new benchmarks that are being added and its more strict about these things.

**Pull request merged**